scripts-dev/update_debian_changelog.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash -e
+# Copyright 2021 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is meant to be used inside a Docker container to run the `dch` incantations
+# needed to release Synapse. This is useful on systems like macOS where such scripts are
+# not easily accessible.
+#
+# Running it (when if the current working directory is the root of the Synapse checkout):
+#   docker run -v $PWD:/synapse ubuntu:latest /synapse/scripts-dev/update_debian_changelog.sh VERSION
+#
+# The image can be replaced by any other Debian-based image (as long as the `devscripts`
+# package exists in the default repository).
+# `VERSION` is the version of Synapse being released without the leading "v" (e.g. 1.42.0).
+
+# Check if a version was provided.
+if [ "$#" -ne 1 ]; then
+  echo "Usage: update_debian_changelog.sh VERSION"
+  echo "VERSION is the version of Synapse being released in the form 1.42.0 (without the leading \"v\")"
```

Comment:
```suggestion
  echo "VERSION is the version of Synapse being released in the form 1.42.0 (without the leading \"v\")"
  echo "This script must be executed in a container where Synapse is at /synapse"
  echo "and apt-get is available to install the `devscripts` package."
```
?

Reply:
That's not what this statement checks. I assume the user has read the instructions on how to run the script, which already mention the image needs to be Debian-based and have the `devscripts` package available.

Reply:
If you assume the user has read the instructions, why are you printing the usage? I'm not sure that's consistent.

Reply:
Because it's imho more likely that the user forgets an argument (or eg passes in the `$ver` variable from the release process but hasn't initialised it correctly) than that they run the script in the wrong context (especially since it'll eventually be just a line to copy/paste from the release instructions).
I'll add extra checks though to make sure apt-get is available and the devscripts package is too.